event_sink_clickhouse/sinks/base_sink.py

```diff
@@ -151,11 +152,14 @@ def get_model(self):
         """
         return get_model(self.model)
 
-    def get_queryset(self):
+    def get_queryset(self, start_pk=None):
         """
         Return the queryset to be used for the insert
         """
-        return self.get_model().objects.all()
+        if start_pk:
+            return self.get_model().objects.filter(pk__gt=start_pk).order_by("pk")
```

Comment:
One last performance improvement we could make here is to use `.values()` or `.values_list()`, which would save a bunch of overhead on creating model instances, instead returning a dict or tuple. `values_list` performs best, but dealing with tuples can be annoying. We could also then choose the fields we want to select to just what we use, potentially saving a lot of data transfer from the db and memory on the job.

Reply:
Dealing with `values_list` would break the serializer, while `values` would return the data already serialized. and we would need to implement a method serializer for every sink. Not sure about the gains here

Reply:
Yeah, I've seen some pretty significant gains in the past from it, but not always. If you're comfortable with the performance as it is we can leave it and revisit later if necessary.